packages/fxa-auth-server/scripts/paypal-processor.ts

```diff
@@ -53,17 +90,42 @@ export async function init() {
   );
   const statsd = Container.get(StatsD);
   statsd.increment('paypal-processor.startup');
-  await processor.processInvoices();
+
+  const timer = initTimer();
+
+  if (useLock) {
+    try {
+      const redis = new Redis(config.redis);
+      const redlock = new Redlock([redis], { retryCount: 1 });
+      lock = await redlock.acquire([program.lockName], lockDuration);
+    } catch (err) {
+      throw new Error(`Cannot acquire lock to run: ${err.message}`);
+    }
+  }
+
+  for await (const _ of processor.processInvoices()) {
+    if (useLock && timer.elapsed() > Math.floor(lockDuration / 2)) {
+      await lock.extend(timer.elapsed());
+      timer.reset();
+    }
+  }
+
+  lock?.release();
   statsd.increment('paypal-processor.shutdown');
   await promisify(statsd.close).bind(statsd)();
   return 0;
 }
 
 if (require.main === module) {
+  let exitStatus = 1;
   init()
+    .then((result) => {
+      exitStatus = result;
+    })
     .catch((err) => {
       console.error(err);
```

Comment:
Sentry capture it?

Reply:
Likely?  The console.error call was already here; I think it's nice to have that output for SREs or just devs running it locally.